src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -840,13 +843,27 @@ private void requestScannedSchematic(@NotNull final Structures.StructureName str
                 Log.getLogger().warn("BuilderTool: server does not have " + serverSideName);
             }
 
-            MineColonies.getNetwork().sendToServer(new BuildToolPlaceMessage(
-                                                                              serverSideName,
-                                                                              structureName.toString(),
-                                                                              Settings.instance.getPosition(),
-                                                                              Settings.instance.getRotation(),
-                                                                              false,
-                                                                              Settings.instance.getMirror()));
+            if(paste)
+            {
+                MineColonies.getNetwork().sendToServer(new BuildToolPasteMessage(
+                        serverSideName,
+                        structureName.toString(),
+                        Settings.instance.getPosition(),
+                        Settings.instance.getRotation(),
+                        false,
+                        Settings.instance.getMirror(),
+                        complete));
```

Comment:
why is here a parameter more than in the message under it?

Reply:
Because only paste needs to differentiate between a complete paste and a nice paste